planner/core/task.go

```diff
@@ -682,6 +683,49 @@ func (p *basePhysicalAgg) newPartialAggregate() (partial, final PhysicalPlan) {
 		groupByItems = append(groupByItems, gbyCol)
 	}
 
+	// Remove unnecessary schema column.
+	// When the select column is same with the group by key, the column can be removed and gets value from the group by key.
+	// e.g
+	// select a, count(b) from t group by a;
+	// The schema is [firstrow(a), count(b), a]. The column firstrow(a) is unnecessary.
+	// Can optimize the schema to [count(b), a] , and change the index to get value.
+	partialCursor = 0
+	partialAggFuncs := make([]*aggregation.AggFuncDesc, 0, len(p.AggFuncs))
+	partialOptimizeSchema := expression.NewSchema()
+	for i, aggFunc := range p.AggFuncs {
+		if aggFunc.Name == ast.AggFuncFirstRow {
+			canOptimize := false
+			for j, gbyExpr := range p.GroupByItems {
+				if gbyExpr.Equal(p.ctx, aggFunc.Args[0]) {
+					canOptimize = true
+					finalAggFuncs[i].Args[0] = groupByItems[j]
+					break
+				}
+			}
+			if canOptimize {
+				partialCursor++
+				continue
+			}
+		}
+		if aggregation.NeedCount(aggFunc.Name) {
+			partialOptimizeSchema.Append(partialSchema.Columns[partialCursor])
+			partialCursor++
+		}
+		if aggregation.NeedValue(aggFunc.Name) {
+			partialOptimizeSchema.Append(partialSchema.Columns[partialCursor])
+			partialCursor++
+		}
+		partialAggFuncs = append(partialAggFuncs, aggFunc)
+	}
+	for i := range p.GroupByItems {
+		partialOptimizeSchema.Append(partialSchema.Columns[partialCursor+i])
+	}
+	p.AggFuncs = partialAggFuncs
+	p.schema = partialOptimizeSchema
```

Comment:
`partialAgg` is a `PhysicalPlan` . It is a `interface`, so it can't use `partialAgg.xxx` to change the value.    line 638 can be removed , and i will update.